server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -697,6 +695,49 @@ protected int cleanupExpiredPoolsImpl() {
         return entitlementsToRegen;
     }
 
+    Set<Pool> revokeEntitlementsFromOverflowingPools(List<Pool> pools) {
+        Collection<Pool> overflowing = new ArrayList<>();
+        for (Pool pool : pools) {
```

Comment:
This would be a really good place to use the stream api.

```
Collection<Pool> overflowing = pools.stream().filter(Pool::isOverflowing).collect(Collectors.toList())
```